src/action.c

```diff
@@ -56,6 +60,52 @@ const char *action_names[] = {
 	NULL
 };
 
+static char *
+action_str_from_arg(struct action_arg *arg)
+{
+	assert(arg->type == LAB_ACTION_ARG_STR);
+	return ((struct action_arg_str *)arg)->value;
+}
+
+static int
+action_int_from_arg(struct action_arg *arg)
+{
+	assert(arg->type == LAB_ACTION_ARG_INT);
+	return ((struct action_arg_int *)arg)->value;
+}
+
+static double
+action_double_from_arg(struct action_arg *arg)
+{
+	assert(arg->type == LAB_ACTION_ARG_DOUBLE);
+	return ((struct action_arg_double *)arg)->value;
+}
+
+static bool
+action_bool_from_arg(struct action_arg *arg)
+{
+	assert(arg->type == LAB_ACTION_ARG_BOOL);
+	return ((struct action_arg_bool *)arg)->value;
+}
+
+static struct wlr_box *
+action_box_from_arg(struct action_arg *arg)
+{
+	assert(arg->type == LAB_ACTION_ARG_BOX);
+	return &((struct action_arg_box *)arg)->value;
+}
```

Comment:
Only `action_str_from_arg()` is used in this PR, the others are examples of how this action_arg framework could be used in the future